app/services/dqt_api.rb

```diff
@@ -44,6 +48,25 @@ def self.get_itt_providers
     response.body["ittProviders"]
   end
 
+  def self.unlock_teacher!(teacher_account)
+    if FeatureFlag.active?(:unlock_teachers_self_service_portal_account)
+      begin
+        response =
+          new.client.put(
+            "/v2/unlock-teacher/#{teacher_account.fetch("uid")}",
+            {}
+          )
+        raise ApiError, response.reason_phrase unless response.success?
+      rescue KeyError,
+             ApiError,
+             Faraday::ConnectionFailed,
+             Faraday::TimeoutError,
+             TooManyResults => e
```

Comment:
I don't think we can get `TooManyResults` from this endpoint:

```suggestion
             Faraday::TimeoutError => e
```

Reply:
Great spot @tvararu . Will update and crack on with unit tests for the `.unlock_teacher!` method